### PR TITLE
General update for 2023-09-01 API

### PR DESCRIPTION
**Pull request by Cyclam**
Additional changes required to work with the 2023-09-01 API.

Reviewing is probably easiest commit by commit. Here they are, with commentary where it might be useful:

- [Update bicep templates to use 2023-09-01](https://github.com/jddarby/azure-cli-extensions/pull/115/commits/c69d31e3177c5e14b181384acee18c56b855fe73): string replace of 2023-01-01-preview for 2023-09-01
- [Update metaschema](https://github.com/jddarby/azure-cli-extensions/pull/115/commits/4131b8a451a52beabb2b77021b604cd35a161167): Use the latest metaschema, including the slight tweak for metaschema_modified.json
- [Add SDK generation instructions to development.md](https://github.com/jddarby/azure-cli-extensions/pull/115/commits/4ec885ae69bbd841d48102b3575e09bd738c62b7): Dev instructions
- [Add Allow-Publisher to required feature flags](https://github.com/jddarby/azure-cli-extensions/pull/115/commits/e515ff1815a2bfca66cfb18bc163bcddfa337e51): 
- [Doc string tidy up](https://github.com/jddarby/azure-cli-extensions/pull/115/commits/ca183a57669798e9a52cdcde583066f7b06172de): 
- [Update to published API (which excludes proxy publisher)](https://github.com/jddarby/azure-cli-extensions/pull/115/commits/9da3ea937bacbdf952deac8c915736b68cc94673): this shouldn't really need reviewing as we're treating the autogenerated code as blackbox (except for the tweak for reversed method parameters). In case you care, this is the SDK regenerated for the final published API. This doesn't include Jacob's flattening changes, as these were already accounted for. 
- [Back out proxy publisher for _get_nfdv()](https://github.com/jddarby/azure-cli-extensions/pull/115/commits/e325e51d52ed176f4d5e7b4b6a8a9e7ce824e1c7): Changes to _our_ code, so does need reviewing
- [Fix tests now proxy publisher has gone](https://github.com/jddarby/azure-cli-extensions/pull/115/commits/adfd3ede5962391eb4e1e985811c339d07427b3b)
- [Add running test help source to development.md](https://github.com/jddarby/azure-cli-extensions/pull/115/commits/97c75eb6ce493d10df9372e3bd3d3bae652d3005)
- [move live tests to uaenorth](https://github.com/jddarby/azure-cli-extensions/pull/115/commits/1712c5f78eec2e62a1846d452c34e2837337b423)
- [Build wheel for 2023-09-01 CLI extension](https://github.com/jddarby/azure-cli-extensions/pull/115/commits/56f356da33d3630755dccf1bb4a41690d2c1d559)
[Update nf_template.bicep.j2 to use new NFDV resource reference property](https://github.com/jddarby/azure-cli-extensions/pull/115/commits/bf60729eb6c750e42c55e9e54c0fd21a0ef6c16a): Fix for bug found by Jen, that the network function API had changed too.

## Testing
Tested CNF and VNF workflow, including SNS create.

### CODE REVIEW

File: src/aosm/azext_aosm/tests/latest/scenario_test_mocks/mock_input_templates/cnf_input_template.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
     "publisher_resource_group_name": "{{publisher_resource_group_name}}",
     "nf_name": "nginx",
     "version": "1.0.0",
-    "acr_artifact_store_name": "nginx-nsd-acr",
-    "location": "westcentralus",
+    "acr_artifact_store_name": "nginx-acr",
+    "location": "uaenorth",
     "images":{
-        "source_registry": "{{source_registry_id}}",
+        "source_registry": "not-used-in-test-but-cannot-be-blank",
```
Comment (jamiedparsons) [inline src/aosm/azext_aosm/tests/latest/scenario_test_mocks/mock_input_templates/cnf_input_template.json:9]: I'm a bit surprised this isn't used anymore.

**Review by jamiedparsons — COMMENTED** [review]
Looks good - a couple of very minor comments.

File: src/aosm/azext_aosm/_client_factory.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
 
 def cf_aosm(cli_ctx, *_) -> HybridNetworkManagementClient:
-    return get_mgmt_service_client(cli_ctx, HybridNetworkManagementClient)
+    return get_mgmt_service_client(cli_ctx, HybridNetworkManagementClient, base_url_bound=False)
```
[the remark this answers is not in the document]
Comment (Cyclam) [reply src/aosm/azext_aosm/_client_factory.py:14]: Done

File: src/aosm/azext_aosm/tests/latest/scenario_test_mocks/mock_input_templates/cnf_input_template.json (hunk 1 above)
Comment (Cyclam) [reply src/aosm/azext_aosm/tests/latest/scenario_test_mocks/mock_input_templates/cnf_input_template.json:9]: I don't know how it used to work. I just know that as it was, `{{source_registry_id}}` evaluated to a blank string (or something that resulted in that), which caused us to fail validation.

I can't add an extensive comment as it's a JSON file. Hopefully the string itself is sufficient for anyone investigating in the future.

**Pull request merged**